### PR TITLE
Issue 596

### LINKED ISSUE #596: Literate Mode NPE
Aya Version: `commit 4406fe4e96b75628857b3ff7347d55d387bf8523`

Minimum code:
```
--| # A
--| A
```

<details>
<summary>Cannot invoke "org.aya.concrete.remark.Literate.resolve(org.aya.resolve.ResolveInfo, org.aya.resolve.context.Context)" because "child" is null</summary>

```
[Info] Resolving source file dependency
In file SF\Basics.aya:1:0 ->

  1 | --| # A
      ^---^
  2 | --| A

Warning: Unsupported markdown syntax: Heading.

    Done in 328ms
  [Tyck] TypeSafeNorm (TypeSafeNorm.aya)
  [Tyck] Structs (Structs.aya)
  [Tyck] StringLit (StringLit.aya)
  [Tyck] Sorts (Sorts.aya)
  [Tyck] SimpleModule (SimpleModule.aya)
  [Tyck] SigmaEta (SigmaEta.aya)
  [Tyck] SF::Basics (SF\Basics.aya)

Cannot invoke "org.aya.concrete.remark.Literate.resolve(org.aya.resolve.ResolveInfo, org.aya.resolve.context.Context)" because "child" is null
java.lang.NullPointerException: Cannot invoke "org.aya.concrete.remark.Literate.resolve(org.aya.resolve.ResolveInfo, org.aya.resolve.context.Context)" because "child" is null
	at org.aya.concrete.remark.Literate$Many.lambda$resolve$0(Literate.java:52)
	at kala.collection.immutable.ImmutableVectors$Vector1.forEach(ImmutableVectors.java:440)
	at org.aya.concrete.remark.Literate$Many.resolve(Literate.java:52)
	at org.aya.concrete.remark.Remark.doResolve(Remark.java:94)
	at org.aya.resolve.visitor.StmtResolver.resolveStmt(StmtResolver.java:49)
	at org.aya.resolve.visitor.StmtResolver.lambda$resolveStmt$0(StmtResolver.java:41)
	at kala.collection.immutable.ImmutableVectors$Vector1.forEach(ImmutableVectors.java:440)
	at org.aya.resolve.visitor.StmtResolver.resolveStmt(StmtResolver.java:41)
	at org.aya.concrete.stmt.Stmt.resolve(Stmt.java:32)
	at org.aya.resolve.module.ModuleLoader.resolveModule(ModuleLoader.java:56)
	at org.aya.cli.library.LibraryModuleLoader.load(LibraryModuleLoader.java:77)
	at org.aya.resolve.module.CachedModuleLoader.lambda$load$0(CachedModuleLoader.java:32)
	at kala.collection.mutable.MutableTreeMap.getOrPut(MutableTreeMap.java:483)
	at org.aya.resolve.module.CachedModuleLoader.load(CachedModuleLoader.java:32)
	at org.aya.resolve.module.ModuleLoader.load(ModuleLoader.java:63)
	at org.aya.cli.library.LibraryCompiler$LibrarySccTycker.tyckOne(LibraryCompiler.java:318)
	at org.aya.cli.library.LibraryCompiler$LibrarySccTycker.tyckSCC(LibraryCompiler.java:305)
	at org.aya.util.tyck.OrgaTycker.tyckSCC(OrgaTycker.java:25)
	at kala.function.CheckedConsumer.accept(CheckedConsumer.java:46)
	at kala.collection.immutable.ImmutableVectors.forEachRec(ImmutableVectors.java:745)
	at kala.collection.immutable.ImmutableVectors$BigVector.forEach(ImmutableVectors.java:567)
	at kala.collection.base.Traversable.forEachChecked(Traversable.java:705)
	at org.aya.cli.library.LibraryCompiler.make(LibraryCompiler.java:208)
	at org.aya.cli.library.LibraryCompiler.make(LibraryCompiler.java:178)
	at org.aya.cli.utils.AyaCompiler.catching(AyaCompiler.java:30)
	at org.aya.cli.library.LibraryCompiler.start(LibraryCompiler.java:143)
	at org.aya.cli.library.LibraryCompiler.compile(LibraryCompiler.java:89)
	at org.aya.test.LibraryTest.compile(LibraryTest.java:110)
	at org.aya.test.LibraryTest.testOnDisk(LibraryTest.java:35)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at jdk.proxy2/jdk.proxy2.$Proxy5.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```

</details>

### PR DESCRIPTION
**Pull request by HoshinoTented**
Try fix #596.

### CODE REVIEW

File: base/src/main/java/org/aya/concrete/remark/Literate.java [hunk 1]
```diff
@@ -135,4 +135,16 @@ public void modify(@NotNull Function<Expr, Expr> fixpoint) {
       });
     }
   }
+
+  record Unsupported(@NotNull ImmutableSeq<Literate> children) implements Literate {
```
Comment (HoshinoTented) [inline base/src/main/java/org/aya/concrete/remark/Literate.java:139]: Maybe `Literate.Many` instead of `ImmutableSeq<Literate>` ?

**Review by ice1000 — APPROVED** [review]
bors r+

**Comment by ice1000** [comment]
```
    java.nio.file.NoSuchFileException: /home/runner/work/aya-dev/aya-dev/base/src/test/resources/literate/test.txt
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:218)
        at java.base/java.nio.file.Files.newByteChannel(Files.java:380)
        at java.base/java.nio.file.Files.newByteChannel(Files.java:432)
        at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:422)
        at java.base/java.nio.file.Files.newInputStream(Files.java:160)
        at java.base/java.nio.file.Files.newBufferedReader(Files.java:2921)
        at java.base/java.nio.file.Files.readAllLines(Files.java:3411)
        at java.base/java.nio.file.Files.readAllLines(Files.java:3452)
        at org.aya.literate.LiterateTest.literate(LiterateTest.java:57)
```

??

**Review by ice1000 — APPROVED** [review]
bors r+

**Comment by ice1000** [comment]
It has to be a bug! I don't understand

File: base/src/main/java/org/aya/concrete/remark/Literate.java [hunk 2]
```diff
@@ -135,4 +135,16 @@ public void modify(@NotNull Function<Expr, Expr> fixpoint) {
...
+    @Override
+    public @NotNull ImmutableSeq<TyckOrder> resolve(@NotNull ResolveInfo info, @NotNull Context context) {
+      return ImmutableSeq.empty();
```
Comment (HoshinoTented) [inline base/src/main/java/org/aya/concrete/remark/Literate.java:142]: Maybe we should resolve children.

Comment (ice1000) [reply base/src/main/java/org/aya/concrete/remark/Literate.java:142]: But they are already unsupported

**Pull request merged**